lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,196 @@
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
+            throws IOException {
+        LogHelper.attachComponentId(configService);
+
+        LOGGER.info(LogHelper.buildLogMessage("Processing report."));
+
+        try {
+            var userIds =
+                    vcStoreItemDataStore.getItems().stream()
+                            .map(VcStoreItem::getUserId)
+                            .distinct()
+                            .toList();
+            List<ReportUserIdentityItem> reportUserIdentities =
+                    userIds.stream()
+                            .map(
+                                    usrId ->
+                                            new ReportUserIdentityItem(
+                                                    usrId,
+                                                    Vot.P0.name(),
+                                                    Collections.emptyList(),
+                                                    false))
+                            .toList();
+            reportUserIdentityDataStore.createOrUpdate(reportUserIdentities);
+
+            processUsersToFindLOCAndUpdateDb(userIds);
+
+            LOGGER.info(
+                    LogHelper.buildLogMessage("Completed report processing for user's identity."));
+        } catch (HttpResponseExceptionWithErrorBody | CredentialParseException | ParseException e) {
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Stopped report processing of user's identity.", e));
+        } catch (BatchDeleteException e) {
+            LOGGER.info(LogHelper.buildLogMessage("Error occurred during batch write process."));
+        } finally {
+            LOGGER.info(LogHelper.buildLogMessage("Write output with result summary."));
+            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+            objectMapper.writeValue(outputStream, buildReportProcessingResult());
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(List<String> userIds)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException {
+        LOGGER.info(
+                LogHelper.buildLogMessage(
+                        String.format("Checking identity for total (%s) users.", userIds.size())));
+        List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
+        for (String userId : userIds) {
+            var tacticalVcs = verifiableCredentialService.getVcs(userId);
+            if (!userIdentityService.areVcsCorrelated(tacticalVcs)) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format("User (%s) VCs not correlated.", userId)));
+                continue;
+            }
+            boolean migrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
```

Comment:
We should probably log a warning here if they don't all match - it would indicate an issue with some of our assumptions.

The only case I can foresee is anyone who completed a F2F journey at the PO but did _not_ return to IPV Core - their F2F VC could have been 'migrated' (as PENDING_RETURN), but the rest weren't. Should be rare/nonexistent though!

Reply:
added a log statement.